src/main.rs

```diff
@@ -218,37 +260,29 @@ fn read_ips_from_file(ips: String) -> Result<Vec<std::net::IpAddr>, std::io::Err
     let mut ips: Vec<std::net::IpAddr> = Vec::new();
 
     for str_ip in reader.lines() {
-        match IpAddr::from_str(&str_ip.unwrap()) {
-            Ok(x) => ips.push(x),
-            Err(y) => panic!("File does not contain valid IP address. Error at {}", y),
-        }
+        // This code assumes that every line in the file is a valid
+        ips.extend(parse_single_ip_or_host(str_ip.unwrap()).unwrap());
     }
     Ok(ips)
 }
 
-fn parse_addresses(opts: &Opts) -> Vec<IpAddr> {
+/// Given a string, parse it as an host, IP address, or CIDR.
+/// This allows us to pass files as hosts or cidr or IPs easily
+/// Call this everytime you have a possible IP_or_host
+fn parse_single_ip_or_host(ip_or_host: String) -> Result<Vec<IpAddr>, std::io::Error> {
```

Comment:
What do you think of calling this `parse_to_ip` and rename the argument to `address`?